Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/ViewBase.cs

```diff
@@ -0,0 +1,291 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace UnityEngine.InputSystem.Editor
+{
+	internal interface IViewStateSelector<out TViewState>
+	{
+		bool HasStateChanged(InputActionsEditorState state);
+		TViewState GetViewState(InputActionsEditorState state);
+	}
+
+	internal interface IView
+	{
+		void UpdateView(InputActionsEditorState state);
+		void DestroyView();
+	}
+
+	internal abstract class ViewBase<TViewState> : IView
+	{
+		public event Action<ViewBase<TViewState>> OnClosing;
+
+		protected ViewBase(StateContainer stateContainer)
+		{
+			this.stateContainer = stateContainer;
+			m_ChildViews = new List<IView>();
+		}
+
+		protected void OnStateChanged(InputActionsEditorState state)
+		{
+			UpdateView(state);
+		}
+
+		public void UpdateView(InputActionsEditorState state)
+		{
+			if (m_ViewStateSelector == null)
+			{
+				Debug.LogWarning($"View '{GetType().Name}' has no selector and will not render. Create a selector for the " +
+				                 $"view using the CreateSelector method.");
+				return;
+			}
+
+			if (m_ViewStateSelector.HasStateChanged(state) || m_IsFirstUpdate)
+				RedrawUI(m_ViewStateSelector.GetViewState(state));
+
+			m_IsFirstUpdate = false;
+			foreach (var view in m_ChildViews)
+			{
+				view.UpdateView(state);
+			}
+		}
+
+		public TView CreateChildView<TView>(TView view) where TView:IView
+		{
+			m_ChildViews.Add(view);
+			return view;
+		}
+
+		public void Close()
+		{
+			OnClosing?.Invoke(this);
+		}
+
+		public void DestroyView<TView>(TView view) where TView : IView
+		{
+			if (view == null)
+				return;
+
+			m_ChildViews.Remove(view);
+			view.DestroyView();
+		}
+
+		public void Dispatch(Command command)
+		{
+			stateContainer.Dispatch(command);
+		}
+
+		public virtual void RedrawUI(TViewState viewState)
+		{
+			
+		}
+
+		/// <summary>
+		/// Called when a parent view is destroying this view to give it an opportunity to clean up any
+		/// resources or event handlers.
+		/// </summary>
+		public virtual void DestroyView()
+		{
+
```

Comment:
Intentionally empty?

Reply:
Would have expected it to call the generic version of it?

Reply:
In this case, yes, since not all views need this. 

> Would have expected it to call the generic version of it?

It's actually the other way around. The generic version calls this one. I've renamed the generic version to DestroyChildViews now, which is a better name for what it does, but we do need to massage this bit of the design a little.